README.md

```diff
@@ -1,95 +1,108 @@
 
-The idea is to provide a community built (and driven) collection of routines that provide ChoiceScript with extended functionality and usage patterns. 
+[ChoiceScript](https://github.com/dfabulich/choicescript) is a simple programming language created by Dan Fabulich to aid non-programmers to write and code interactive fiction text-based gamebooks in the style of [Choose Your Own Adventure](https://en.wikipedia.org/wiki/Choose_Your_Own_Adventure). The language is proprietary and governed by its own [license](https://github.com/dfabulich/choicescript/blob/master/LICENSE.txt).
 
-- **Unblock Creation**
-    cslib can aid in ChoiceScript development by providing complicated functionality out of the box, or allowing certain code patterns to be written more concisely.
+The idea is to provide a community built (and driven) collection of routines, so [contribute](#-contributing) with ideas, code or feedback!
 
-- **Stimulate Learning**
-    cslib is written in ChoiceScript, so it's easy for any ChoiceScript author to read, understand, and learn from the cslib code. In time, they may even choose to contribute themselves.
+**Create, Learn and Experiment!**
 
-- **Encourage Experimentation**
-    This one needs little explanation. Whether you're helping create cslib, just using it your games, all that matters is that you're enjoying the experience.
+## Use It!
 
-## Overview 
+### Installation
+- Copy the file to your game folder together with your scene files.
 
-cslib is organized into 'modules' (scene files) that represent certain categories of functionality, for example, string (cslib_string), or number manipulation (cslib_number).
+- Create a global variable called `cslib_ret` to hold return values.
 
-### Usage
+```choicescript
+__startup.txt__
 
-To use cslib, you need only include the cslib_ prefixed scene files in your project, and define a few configuration variables in your startup scene (details further below). You can then call each individual routine as required, via gosub_scene. Documentation on each routine, including its functionality and required parameters (arguments) can be found in the commented source files.
+*create cslib_ret 0
+```
 
-### Limitations
+- That's it! :partying_face:
 
-Sadly, cslib is not always going to be a one-size-fits all solution to your ChoiceScript woes. There are inherent limitations in ChoiceScript that cannot be overcome, no matter how much extra ChoiceScript we write. In addition, when writing a utility like cslib (that aims to appeal to a large number of people), certain tradeoffs have to be made. So, if you plan to use it, you should keep the following limitations in mind:
+You can now call any of the routines described in the file. :muscle:
```

Comment:
file**S**

Also worth clarifying "public" routines? Any labels that are **not** preceded with an underscore.